### PR TITLE
support select command #46

### LINKED ISSUE #46: support select command
`select #` scrolls and focuses on the specified item (which also loads the corresponding webpage).

### PR DESCRIPTION
**Pull request by lejolly**
Fixes #46

### CODE REVIEW

File: src/main/java/seedu/address/parser/Parser.java [hunk 1]
```diff
@@ -167,6 +178,20 @@ private Command prepareDelete(String args) {
     }
 
     /**
+     * Re-direct to prepareSelectOrDelete method
```
Comment (damithc) [inline src/main/java/seedu/address/parser/Parser.java]: Re-directs

File: src/main/java/seedu/address/parser/Parser.java [hunk 2]
```diff
@@ -146,19 +150,26 @@ private static UniqueTagList getTagsFromArgs(String tagArguments) throws Illegal
...
         try {
             final int targetIndex = parseArgsAsDisplayedIndex(args);
-            return new DeletePersonCommand(this.ui.getDisplayedPersons().get(targetIndex - 1));
+            // this check is required for both delete and select to catch any invalid indexes
+            ReadOnlyPerson person = this.ui.getDisplayedPersons().get(targetIndex - 1);
+            if (isDelete) {
+                return new DeletePersonCommand(person);
+            } else {
+                return new SelectPersonCommand(targetIndex);
+            }
```
Comment (damithc) [inline src/main/java/seedu/address/parser/Parser.java]: `return isDelete ? ... : ...` ?

File: src/main/java/seedu/address/parser/Parser.java [hunk 3]
```diff
@@ -146,19 +150,26 @@ private static UniqueTagList getTagsFromArgs(String tagArguments) throws Illegal
...
-     * Parses arguments in the context of the delete person command.
+     * Parses arguments in the context of the select or delete person command.
      *
      * @param args full command args string
+     * @param isDelete determines if returned command is a SelectPersonCommand or a DeletePersonCommand
      * @return the prepared command
      * @throws ParseException containing a message with relevant info if the args could no be parsed
      */
-    private Command prepareDelete(String args) {
+    private Command prepareSelectOrDelete(String args, boolean isDelete) {
```
Comment (damithc) [inline src/main/java/seedu/address/parser/Parser.java]: Merging these two methods is a bit dodgy. It minimizes code but the abstraction is not intuitive. If there is no common abstraction we can extract, i'd rather have two similar methods.

Comment (lejolly) [inline src/main/java/seedu/address/parser/Parser.java]: fixed.

**Comment by damithc** [comment]
No comments from me.

**Comment by damithc** [comment]
@yl-coder can review?

File: src/test/java/guitests/PersonSelectCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    @Test
+    public void selectPerson_middlePerson_successful() {
+        int numberOfPeople = personListPanel.getNumberOfPeople();
+        assertTrue(numberOfPeople > 5);
+        assertEquals(personListPanel.getSelectedPersons().size(), 0);
+
+        personListPanel.enterCommandAndApply("select 3");
+        assertEquals(personListPanel.getSelectedPersons().size(), 1);
+        ReadOnlyPerson thirdPerson = personListPanel.getSelectedPersons().get(0);
+        assertEquals(personListPanel.getPerson(2), thirdPerson);
```
Comment (yl-coder) [inline src/test/java/guitests/PersonSelectCommandTest.java]: We could check the statusBar if it is displaying the correct message also. If any.

**Comment by yl-coder** [comment]
No further comment from me.

File: src/test/java/guitests/PersonSelectCommandTest.java (hunk 4 above)
Comment (lejolly) [inline src/test/java/guitests/PersonSelectCommandTest.java]: got it.

**Pull request merged**